ui/appDisplay.js

```diff
@@ -102,20 +96,30 @@ function createInfoPopup() {
     Main.uiGroup.add_actor(this._infoPopup.actor);
 
     this.track_hover = true;
-    this.connect('notify::hover', () => {
+    return this.connect('notify::hover', () => {
         if (this.hover) {
             this._infoPopupId = GLib.timeout_add(
```

Comment:
I think you also want to remove the old timeout source here before setting up a new one.

Reply:
Probably worth to also connect to popup menu destroy and disconnect this timeout source there.

Reply:
Done